app/grandchallenge/algorithms/tasks.py

```diff
@@ -95,26 +36,19 @@ def execute_algorithm_job_for_inputs(*, job_pk):
         kwargs={"job_pk": str(job.pk)}, immutable=True
     )
 
-    # check if all ComponentInterfaceValue's have a value.
-    missing_inputs = list(civ for civ in job.inputs.all() if not civ.has_value)
-
-    if missing_inputs:
-        job.update_status(
-            status=job.CANCELLED,
-            error_message=(
-                f"Job can't be started, input is missing for "
-                f"{oxford_comma([c.interface.title for c in missing_inputs])}"
-            ),
-        )
-        on_commit(linked_task.apply_async)
-    else:
+    if job.inputs_complete:
         job.task_on_success = linked_task
         job.save()
         on_commit(
             execute_algorithm_job.signature(
                 kwargs={"job_pk": job_pk}, immutable=True
             ).apply_async
         )
+    else:
```

Comment:
What happens if the any of the preceding tasks fail, is the job left in a strange state?

Reply:
By fail, do you mean unexpectedly fail with an error that we don't catch in the task or System failure? If any of the previous tasks fail due to unexpected errors, the follow-up tasks won't be called and the job will indeed be left dangling. The solution to that would be to catch any exception in the task and update the object and re-raise the exception, right?

Reply:
Yes I think so.

Reply:
There must a better way. If a task fails, you would want some task_on_failure to run that updates the job objects status, right? Looking at the celery docs, that should be possible using a chord with an error callback, but I think we've been having lots of issues with celery chords or any advanced canvas stuff, so probably not the way to go.

Reply:
The celery stuff is unreliable. You first proposal is exactly how we solve it in other tasks.